options.go

```diff
@@ -0,0 +1,85 @@
+package jrpc
+
+import (
+	"net/http"
+	"time"
+)
+
+// Option func type
+type Option func(s *Server)
+
+// Auth sets basic auth credentials, required
+func Auth(user, password string) Option {
+	return func(s *Server) {
+		s.authUser = user
+		s.authPasswd = password
+	}
+}
+
+// AppName sets custom application name for server signature, optional
+func AppName(name string) Option {
+	return func(s *Server) {
+		s.appName = name
+	}
+}
+
+// Version sets app version value for server signature, optional
+func Version(version string) Option {
+	return func(s *Server) {
+		s.version = version
+	}
+}
+
+// Timeouts sets server timeout values such as ReadHeader, Write and Idle timeout, optional.
+// If this option not defined server use default timeout values
+func Timeouts(limits Limits) Option {
+
+	// this option sets only server limits values and exclude middlewares limits fields of Limits struct.
+	return func(s *Server) {
+		s.limits.ReadHeaderTimeout = limits.ReadHeaderTimeout
+		s.limits.WriteTimeout = limits.WriteTimeout
+		s.limits.IdleTimeout = limits.IdleTimeout
+	}
+}
+
+// WithLimits sets value for client limit call/sec per client middleware
+func WithLimits(limit float64) Option {
+
+	// this option sets only server limits values and exclude middlewares limits fields of Limits struct.
+	return func(s *Server) {
+		s.limits.clientLimit = limit
+	}
+}
+
+// WithThrottler sets throttler middleware with specify limit value, optional
+func WithThrottler(limit int) Option {
+	return func(s *Server) {
+		s.limits.serverThrottle = limit
+	}
+}
+
+// WithTimeout sets middleware with custom timeout value, optional
+func WithTimeout(timeout time.Duration) Option {
```

Comment:
this is confusing name as we already have Timeouts option. Maybe this on should moved to limits structure and set by the same Timeouts

Reply:
Yes. I thought about this and it doesn't like me too. I will move it to `Timeouts`